manager/allocator/allocator.go

```diff
@@ -83,74 +57,11 @@ func New(store *store.MemoryStore, pg plugingetter.PluginGetter) (*Allocator, er
 func (a *Allocator) Run(ctx context.Context) error {
 	// Setup cancel context for all goroutines to use.
 	ctx, cancel := context.WithCancel(ctx)
-	var (
-		wg     sync.WaitGroup
-		actors []func() error
-	)
 
 	defer func() {
 		cancel()
-		wg.Wait()
 		close(a.doneChan)
 	}()
-
-	for _, aa := range []allocActor{
-		{
-			taskVoter: networkVoter,
-			init:      a.doNetworkInit,
-			action:    a.doNetworkAlloc,
-		},
-	} {
-		if aa.taskVoter != "" {
-			a.registerToVote(aa.taskVoter)
-		}
-
-		// Assign a pointer for variable capture
-		aaPtr := &aa
```

Comment:
Worth noting, this literally never would have worked, because the value of aaPtr is the same every iteration.

https://tam7t.com/golang-range-and-pointers/